pilot/pkg/model/push_context.go

```diff
@@ -1044,6 +1044,16 @@ func (ps *PushContext) DelegateVirtualServices(vses []config.Config) []ConfigHas
 // Callers can check if the sidecarScope is from user generated object or not
 // by checking the sidecarScope.Config field, that contains the user provided config
 func (ps *PushContext) getSidecarScope(proxy *Proxy, workloadLabels labels.Instance) *SidecarScope {
+	sidecar := ps.doGetSidecarScope(proxy, workloadLabels)
+	// we need to make sure sidecar scope is initiaized before returning
+	if features.EnableLazySidecarEvaluation && sidecar != nil && sidecar.initFunc != nil {
+		sidecar.initFunc()
```

Comment:
I believe if a sidecar is namespace scoped, it can cause race here

Reply:
Not sure which race could occur, `initFunc` uses `sync.OnceFunc` if multiple proxies attempt to initialize the same sidecar they would be blocked until initialization is completed. this should not be a mayor issue because initialization should be very quick for each proxy.

Reply:
hmm, will double check

Reply:
right, it uses sync once